src/chainstate/stacks/db/unconfirmed.rs

```diff
@@ -157,6 +162,8 @@ impl UnconfirmedState {
                 &self.confirmed_chain_tip,
             );
 
+            clarity_tx.reset_cost(cur_cost);
```

Comment:
Top of mind, so sharing, but I think I remember having some issues a few months ago with this function, not doing what I was expecting, were you able to get the results you were looking for?

Reply:
I'm not sure I know what this is?  @kantai do you?

Reply:
No -- `reset_cost` should work to set the cost tracker's current cost, and the implementation is pretty explicit.